cmd/router/key.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Stichting ThingsIX Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"github.com/ThingsIXFoundation/packet-handling/router"
+	"github.com/spf13/cobra"
+)
+
+var keyCmd = &cobra.Command{
+	Use:   "key",
+	Short: "key commands",
+}
+
+var genKeyCmd = &cobra.Command{
+	Use:   "generate <keyfile>",
```

Comment:
Since a config is required anyway to run this command. Get the keyfile from the config or drop the config requirement.

Reply:
It isn't required for this subcommand. This just generates an router identity key in the given path. Configuration is only required when the router is run as a service.

We can make it optional. If the user gives a configuration it uses the keyfile path from the config and optionally overwrites it if an explicit path is provided.

Reply:
Let's do that